### DISCUSSION TITLE
How to add to multiple models?

### DISCUSSION THREAD

**Discussion by tobz-nz** [Z3d0X/filament-fabricator#44] [Ideas]
Sorry Discussions are disabled so asking this question here instead.

I can't figure out how to add fabricator to my other models - is it fixed to only the provided Pages model?
I really want to use it on a number of different models - is this possible?

**Comment by Z3d0X** [comment]
https://github.com/Z3d0X/filament-fabricator/issues/26#issuecomment-1321486762

**Comment by tobz-nz** [comment]
I've worked out you can add blocks to any Model.

```php
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput:make('title'),
                Group::make()
                    ->schema([
                        PageBuilder::make('blocks')
                            ->label(__('filament-fabricator::page-resource.labels.blocks'))
                            ->blocks(FilamentFabricator::getPageBlocks()),
                    ])->columnSpan(2),
            ]);
    }
```

And add a json field called `blocks to your table.

```php
$table->json('blocks')->nullable();
```

Seems to work fine, so..? 🤷‍♂️

**Reply by luckykenlin** [reply]
Save me alot time. thanks

**Comment by Z3d0X** [comment]
Awesome, that works for you! I originally had a different idea than this. Thanks for sharing your idea
